Wikipedia/UI-V5/WMFTableOfContentsViewController.swift

```diff
@@ -95,19 +95,23 @@ public class WMFTableOfContentsViewController: UITableViewController, WMFTableOf
         }
     }
 
+    // MARK: - Header
+    func forceUpdateHeaderFrame(){
+        //See reason for fix here: http://stackoverflow.com/questions/16471846/is-it-possible-to-use-autolayout-with-uitableviews-tableheaderview
+        self.tableView.tableHeaderView!.setNeedsLayout()
+        self.tableView.tableHeaderView!.layoutIfNeeded()
+        let headerHeight = self.tableView.tableHeaderView!.systemLayoutSizeFittingSize(UILayoutFittingCompressedSize).height
+        var headerFrame = self.tableView.tableHeaderView!.frame;
+        headerFrame.size.height = headerHeight
+        self.tableView.tableHeaderView!.frame = headerFrame;
+        self.tableView.tableHeaderView = self.tableView.tableHeaderView
```

Comment:
@bgerstle yeah - thats how I feel about it too… basically you need to manually update the header view to get the sizing right and then reset the tableHeaderView property to make the table view respect the new size. 

From my poking around and SO spelunking this seems to be a known problem, and this answer was accepted a few times on variations of this question. 

I think we have seen this before - that headers and footers don't necessarily play nice with auto layout.